From 6bbec909f30b7e2356a08c7a7e4fdc46ae11ef05 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Thu, 9 Feb 2012 10:05:15 +0100
Subject: [PATCH] systemctl: check for no more work after chkconfig

Avoid a bogus message from 'systemctl enable ...' when all units given
are SysV services:
   Warning: unit files do not carry install information. No operation
   executed.

https://bugzilla.redhat.com/show_bug.cgi?id=787407
(cherry picked from commit ab5919fa1af147c2632fdae7bed4504c898a60af)
---
 src/systemctl.c |    7 +++++--
 1 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/systemctl.c b/src/systemctl.c
index c86ec80..8c94fd3 100644
--- a/src/systemctl.c
+++ b/src/systemctl.c
@@ -3643,12 +3643,15 @@ static int enable_unit(DBusConnection *bus, char **args) {
         int r;
         DBusError error;
 
-        dbus_error_init(&error);
-
         r = enable_sysv_units(args);
         if (r < 0)
                 return r;
 
+        if (!args[1])
+                return 0;
+
+        dbus_error_init(&error);
+
         if (!bus || avoid_bus()) {
                 if (streq(verb, "enable")) {
                         r = unit_file_enable(arg_scope, arg_runtime, arg_root, args+1, arg_force, &changes, &n_changes);
